### PR TITLE
Allow local configs

### PR DESCRIPTION
**Pull request by McPolemic**
Users are going to have different usernames, different passwords, different database setups. By having an untracked `config/local.exs`, we can allow users to override settings without having to change version-controlled files like `config/dev.exs`.

I've also added a `config/local.exs.example` that can be copied over to start configuring an install.

### CODE REVIEW

**Comment by RBangel** [comment]
Still undecided, but this feels like the wrong solution. The dev, test, and prod files should be changed for your environment.  Maybe that means they should be built around env variables or maybe that means they should have .examples instead and aren't checked in... we shouldn't have a single global override.

**Comment by McPolemic** [comment]
I see where you're coming from but I'm not entirely sure I agree. We may set environment-specific configuration in those files and we can't rely on making changes to example files and hoping that those changes trickle back into untracked config files. I've seen this model of having an ignored local config and other projects and I think it's a good compromise between install customizability and the ability to set environment-specific configuration for project.

File: config/config.exs [hunk 1]
```diff
@@ -29,3 +29,6 @@ config :we_need_to_watch, :basic_auth,
 # Import environment specific config. This must remain at the bottom
 # of this file so it overrides the configuration defined above.
 import_config "#{Mix.env}.exs"
+
+# Override with local configuration (gitignored)
+import_config "local.exs"
```
Comment (RBangel) [inline config/config.exs]: I think I'd like it to be conditional... use it if you want, don't if you don't
```
if File.exists?("./local.exs"), do: import_config "local.exs"
```

Thoughts?

Comment (McPolemic) [reply config/config.exs]: I'm fine with that. Should I also pull out the hard-coded `raelyn` user in `config/config.exs` and replace it with the traditional `postgres` user?

Comment (RBangel) [reply config/config.exs]: I don't see that in `config/config.exs`... We should remove it from `config/dev.exs` if that is the one you meant.

**Pull request merged**